tempesta_fw/http_parser.c

```diff
@@ -1842,21 +1871,21 @@ tfw_http_parse_req(void *req_data, unsigned char *data, size_t len)
 		if (unlikely(!IN_ALPHABET(c, hdr_a)))
 			return TFW_BLOCK;
 
-		tfw_http_msg_hdr_open(msg, p);
+		__msg_hdr_open(p);
 
 		switch (LC(c)) {
 		case 'c':
 			__FSM_MOVE(Req_HdrC);
 		case 'h':
-			if (likely(p + 5 <= data + len
+			if (likely(__data_available(p, 5)
```

Comment:
It might be a good idea to use real strings in cases like this. That would make the code more readable. Say, in this particular case we could do this:
```C
#define SLEN(s)    (sizeof(s) - 1)
if (likely(__data_available(p, SLEN("Host:"))
             && C4_INT_LCM(p + 1, 'o', 's', 't', ':'))) {
        parser->_i_st = Req_HdrHostV;
        parser->_hdr_tag = TFW_HTTP_HDR_HOST;
        __FSM_MOVE_n(RGen_LWS_empty, SLEN("Host:"));
}
```
We use `SLEN()` macro in other parts of Tempesta code in similar fashion, even though, I believe, it's always local to a module.